test/integration/test_sriov_cni.sh

```diff
@@ -0,0 +1,109 @@
+#!/bin/bash
+
+this_folder="$(dirname "$(readlink --canonicalize "${BASH_SOURCE[0]}")")"
+export CNI_PATH=${this_folder}
+export CNI_CONTAINERID=container_x
+
+
+setup() {
+  ip netns del test_root_ns || true
+  ip netns add test_root_ns
+
+  # See pkg/utils/testing.go
+  ip netns exec test_root_ns ip link add enp175s0f1 type dummy
+  ip netns exec test_root_ns ip link add enp175s6 type dummy
+  ip netns exec test_root_ns ip link add enp175s7 type dummy
+
+  DEFAULT_CNI_DIR=$(mktemp -d)
+  export DEFAULT_CNI_DIR
+}
+
+
+test_macaddress() {
+
+  make_container "container_1"
+
+  export CNI_NETNS=/run/netns/container_1_netns
+  export CNI_IFNAME=net1
+
+  read -r -d '' CNI_INPUT <<- EOM
+  {
+    "type": "sriov",
+    "cniVersion": "0.3.1",
+    "name": "sriov-network",
+    "ipam": {
+      "type": "test-ipam-cni"
+    },
+    "deviceID": "0000:af:06.0",
+    "mac": "60:00:00:00:00:E1",
+    "logLevel": "debug"
+  }
+EOM
+
+  export CNI_COMMAND=ADD
+  assert invoke_sriov_cni
+  assert 'ip netns exec container_1_netns ip link | grep -i 60:00:00:00:00:E1'
+
+  export CNI_COMMAND=DEL
+  assert 'invoke_sriov_cni'
+  assert 'ip netns exec test_root_ns ip link show enp175s6'
+}
+
+
+test_vlan() {
+
+  make_container "container_1"
+
+  export CNI_NETNS=/run/netns/container_1_netns
+  export CNI_IFNAME=net1
+
+  read -r -d '' CNI_INPUT <<- EOM
+  {
+    "type": "sriov",
+    "cniVersion": "0.3.1",
+    "name": "sriov-network",
+    "vlan": 1234,
+    "ipam": {
+      "type": "test-ipam-cni"
+    },
+    "deviceID": "0000:af:06.0",
+    "mac": "60:00:00:00:00:E1",
+    "logLevel": "debug"
+  }
+EOM
+
+  export CNI_COMMAND=ADD
+  assert invoke_sriov_cni
+  assert_file_contains "${DEFAULT_CNI_DIR}/enp175s0f1.calls" "LinkSetVfVlanQosProto enp175s0f1 0 1234 0 33024"
+
```

Comment:
I think we need to remove this file after every test no?

Reply:
it's not mandatory, because the `setup()` function creates a different DEFAULT_CNI_DIR at every test:
```
setup() {
  ...
  DEFAULT_CNI_DIR=$(mktemp -d)
  export DEFAULT_CNI_DIR
}
```

But it is still a good practice. Adding a `teardown()` function